[DOC] Documented ax argument and the figure in plot_series

ShreeshaM07 opened the pull request:
#### Reference Issues/PRs

#### What does this implement/fix? Explain your changes.
This resolves #5323 . I have added the `ax` argument and the figure to the docstring of the plot_series.

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### What should a reviewer concentrate their feedback on?
    If the docstring meets the requirements.

#### Did you add any tests for the change?
No

#### Any other comments?
Since I am new to open source and this being my first pull request any feedback would be helpful.

#### PR checklist
- [X] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

fkiraly reviewed (approved):
Thanks, much improved!

It does not resolve #5323 entirely though, as the resulting figure is not described yet. Still happy to merge though as it solves half of the issue.

Non-blocking, the description of `fig` sounds a bit strange but is not strictly speaking wrong, perhaps you could look over it to make the description a bit clearer?

fkiraly commented:
Oh, and I think you have accidentally introduced trailing whitespaces, see here how to avoid these: https://www.sktime.net/en/stable/developer_guide/coding_standards.html

ShreeshaM07 commented:
Regarding the trailing whitespaces. I gave 4 whitespaces instead of a tab space before each line, is there anything else that I should be doing to avoid the trailing whitespaces.

fkiraly commented:
> Regarding the trailing whitespaces. I gave 4 whitespaces instead of a tab space before each line, is there anything else that I should be doing to avoid the trailing whitespaces.

The true secret is, if there is an entirely empty line, it should not start with 4 whitespaces but zero whitespaces. Some may say it is inconsistent, but that's how the linter wants it.

ShreeshaM07 commented:
Sorry was busy for the past couple of days. I have now added the description for the plotted figure and also removed the trailing white spaces. If there is anything else to be resolved please let me know.

fkiraly reviewed (changes requested):
Thanks a lot!

For the description of the figure, may I suggest to:

* move it to the top
* make references to the actual parameters
* also give a description of the figure, i.e., "x-axis is time, y-axis is value of the time series. All time series in `series` are plotted into the figure, with different markers and colors. Custom markers and colors can be passed via `markers` and `colors`." and so on
* avoid valuative statements like the "it is a versatile tool" and restrict yourself to a description

fkiraly commented on sktime/utils/plotting.py:
```diff
@@ -30,6 +30,22 @@ def plot_series(
 ):
     """Plot one or more time series.
 
+    Description
```
Thanks!

can you kindly remove the "description" header, this is not an official numpydoc header.

fkiraly reviewed (changes requested):
Excellent, almost ready to go!

Only blocking comment is removing the "description" header.

ShreeshaM07 commented:
there was an issue while pushing the changes will be doing it in another pull request.

fkiraly reviewed (changes requested):
The changes to `plotting` are fine, although it looks like you accidentally pushed changes to `01a_forecasting_sklearn.ipynb` with your latest commit. Can you kindly remove these from the PR?

An easy way to do this is to copy the original version of the file from `main` over it and commit that to the branch you are working on. As the PR merge will squash all changes, this will cancel out and result in no activity on the affected file.

ShreeshaM07 commented:
I copied the code and re-ran the `01a_forecasting_sklearn.ipynb` but still it is not giving the same execution_count values as was present before I changed this file by mistake initially. Is there anyway I can get the execution_count values to match with what is present in the original source code.

fkiraly commented:
It seems like they match?

ShreeshaM07 commented:
Has the changes made to `plotting.py` been updated?

fkiraly reviewed (approved):
Thanks, all good now!
